Neos.ContentRepository.Core/Classes/Feature/WorkspaceCommandHandler.php

```diff
@@ -504,6 +505,13 @@ private function handlePublishIndividualNodesFromWorkspace(
         $workspace = $this->requireWorkspace($command->workspaceName, $contentRepository);
         $baseWorkspace = $this->requireBaseWorkspace($workspace, $contentRepository);
 
+        // 0) close old content stream
+        $contentRepository->handle(
+            CloseContentStream::create(
+                $workspace->currentContentStreamId,
+            )
+        );
```

Comment:
Below (in line 605ff) we trigger a `RemoveContentStream` event:
```php
        // to avoid dangling content streams, we need to remove our temporary content stream (whose events
        // have already been published)
        $contentRepository->handle(RemoveContentStream::create(
            $matchingContentStream
        ));
```

So we already have a "final" event on the CS.. Maybe we can trigger the `ContentStreamWasRemoved` event _first_ but since we rely on the read model for the "hard constraint" checks during rebasing, those would fail if we changed the order..